src/Symfony/Bridge/Doctrine/Security/User/EntityUserProvider.php

```diff
@@ -54,8 +54,17 @@ public function loadUserByUsername($username)
         if (null !== $this->property) {
             $user = $this->repository->findOneBy(array($this->property => $username));
         } else {
-            if (!$this->repository instanceof UserProviderInterface) {
-                throw new \InvalidArgumentException(sprintf('The Doctrine repository "%s" must implement UserProviderInterface.', get_class($this->repository)));
+            if (!($this->repository instanceof UserLoaderInterface
+                || $this->repository instanceof UserProviderInterface)
+            ) {
+                throw new \InvalidArgumentException(sprintf(
+                    'The Doctrine repository "%s" must implement UserLoaderInterface.',
+                    get_class($this->repository)
+                ));
+            }
+
+            if ($this->repository instanceof UserProviderInterface) {
+                @trigger_error('Implementing loadUserByUsername from Symfony\Component\Security\Core\User\UserProviderInterface is deprecated since version 2.7 and will be removed in 3.0. Use the Symfony\Bridge\Doctrine\Security\User\UserLoaderInterface instead.', E_USER_DEPRECATED);
```

Comment:
This should be triggered only if the class does not implement ``UserLoaderInterface``, to be able to implement both interfaces in case you need to keep BC with older versions